runtime/parachains/src/disputes/migration.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Storage migration(s) related to disputes pallet
+
+use frame_support::traits::StorageVersion;
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+pub mod v1 {
+	use super::*;
+	use crate::disputes::{Config, Pallet};
+	use frame_support::{
+		pallet_prelude::*, storage_alias, traits::OnRuntimeUpgrade, weights::Weight,
+	};
+	use primitives::v2::SessionIndex;
+	use sp_std::prelude::*;
+
+	#[storage_alias]
+	type SpamSlots<T: Config> = StorageMap<Pallet<T>, Twox64Concat, SessionIndex, Vec<u32>>;
+
+	pub struct MigrateToV1<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let mut weight: Weight = Weight::zero();
+
+			if StorageVersion::get::<Pallet<T>>() < STORAGE_VERSION {
+				log::info!(target: crate::disputes::LOG_TARGET, "Migrating disputes storage to v1");
+				weight += migrate_to_v1::<T>();
+				STORAGE_VERSION.put::<Pallet<T>>();
+				weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
```

Comment:
q: Why are we adding to the `weight` here? Is it for reading and writing the storage version? I noticed that in `configuration/migration.rs` we don't do this.

Reply:
As @ordian explained it [here](https://github.com/paritytech/polkadot/pull/6345#discussion_r1050796336), adding read/write weight for the storage version is expected but not that important, nothing will explode if you forget to add it.

Reply:
I see! Thanks.